src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransPatterns.java

```diff
@@ -406,27 +403,33 @@ private void handleSwitch(JCTree tree,
             Assert.check(preview.isEnabled());
```

Comment:
side: not part of your patch but, do we need to check if preview is enabled at this point?